backend/db/resolve_fetch_failures_0_members.py

```diff
@@ -2,46 +2,124 @@
+def _report_success(
+    cset_ids: List[int], fetch_audit_row_lookup: Dict[str, Dict], comment_addition: str = None, use_local_db=False
+):
+    """Report success for a list of concept set IDs.
+    todo: kind of weird to pass a lookup. Maybe do that beforehand and pass row dict"""
+    success_rows = []
+    for cset_id in cset_ids:
+        row = fetch_audit_row_lookup[str(cset_id)]
+        if comment_addition:
+            row['comment'] = row['comment'] + '; ' + comment_addition
+        success_rows.append(row)
+    fetch_status_set_success(success_rows, use_local_db)
 
-def resolve_fetch_failures_0_members(version_id: int = None, use_local_db=False):
+def resolve_fetch_failures_0_members(version_id: int = None, use_local_db=False, polling_interval_seconds=30):
     """Resolve situations where we tried to fetch data from the Enclave, but failed due to the concept set being too new
     resulting in initial fetch of concept set members being 0.
     :param version_id: Optional concept set version ID to resolve. If not provided, will check database for flagged
-    failures."""
-    print('Resolving fetch failures: new concept set / 0 members')
-    print('id:')
-    print(version_id)
-    if version_id:
-        print(f'Fetching version ID: {version_id}')
-        with get_db_connection(local=use_local_db) as con:
-            pass
-            # refresh_termhub_core_cset_derived_tables(con, SCHEMA)
-
-
-# todo: Ideally allow for output_dir for testing purposes etc, but datasets.py currently supports only 1 of 2 outdirs
-#  needed. See 'todo' above its cli() func.
+    failures.
+    todo: Performance: Fetch only members, ideally: Even though we are fetching 'members', adding to the cset members
+     table requires cset version and container metadata, and the function that does this expects them to be formaatted
+     as objects, not as they come from our DB. We can fetch from DB and then convert to objects, but a lot of work for
+     small performance gain."""
+    print("Resolving fetch failures: non-draft, osensibly new concept sets with >0 expressions but 0 members")
+    # Collect failures
+    failures: List[Dict] = select_failed_fetches(use_local_db)
+    failure_lookup: Dict[str, Dict] = {x['primary_key']: x for x in failures}
+    failed_cset_ids: List[int] = [version_id] if version_id else [
+        int(x['primary_key']) for x in failures if x['status_initially'] == 'fail-0-members']
+    failed_cset_ids = list(set(failed_cset_ids))  # dedupe
+    if not failed_cset_ids:
+        print("No failures to resolve.")
+        return
+
+    # Resolve
+    t0 = datetime.now()
+    print(f"Fetching concept set versions and their related objects: {', '.join([str(x) for x in failed_cset_ids])}")
+    with get_db_connection(schema='', local=use_local_db) as con:
+        i = 0
+        while len(failed_cset_ids) > 0 and (datetime.now() - t0).total_seconds() < 2 * 60 * 60:  # 2 hours
```

Comment:
## Handling repeated fetches for duration
@Sigfried I know that you suggested using the existing functionality of `make_object_request()`. I was going to include that for some of these cases. However, I felt it a good idea to set the script up so that it can handle multiple failure cases (any that show up in the `fetch_audit` table), even though you can call it on a single cset. So because of multiple possible failures, I didn't want to use `make_object_request()` because that would stay stuck on 1 cset at a time.